apps/managedidentity/managedidentity_test.go

```diff
@@ -217,6 +281,293 @@ func Test_SystemAssigned_Returns_Token_Success(t *testing.T) {
+	setCustomAzureArcPlatformPath(t, testCaseFilePath)
+
+	createMockFile(t, mockFilePath, 0)
+
+	headers := http.Header{}
+	headers.Set(wwwAuthenticateHeaderName, basicRealm+filepath.Join(testCaseFilePath, secretKey))
+
+	mockClient.AppendResponse(mock.WithHTTPStatusCode(http.StatusUnauthorized),
+		mock.WithHTTPHeader(headers),
+		mock.WithCallback(func(r *http.Request) {
+			localUrl = r.URL
+		}))
+
+	responseBody, err := getSuccessfulResponse(resource)
+	if err != nil {
+		t.Fatalf(errorFormingJsonResponse, err.Error())
+	}
+	// adding success response.
+	mockClient.AppendResponse(mock.WithHTTPStatusCode(http.StatusOK), mock.WithHTTPHeader(headers),
+		mock.WithBody(responseBody), mock.WithCallback(func(r *http.Request) {
+			localUrl = r.URL
+		}))
+
+	// resetting cache
+	before := cacheManager
+	defer func() { cacheManager = before }()
+	cacheManager = storage.New(nil)
+
+	client, err := New(SystemAssigned(), WithHTTPClient(&mockClient))
+	if err != nil {
+		t.Fatalf(`error not expected, got error: "%s"`, err)
+	}
+	result, err := client.AcquireToken(context.Background(), resource)
+	if err != nil {
+		t.Fatalf(`error not expected, got error: "%s"`, err)
+	}
+
+	if localUrl == nil || !strings.HasPrefix(localUrl.String(), endpoint) {
+		t.Fatalf("url request is not on %s got %s", endpoint, localUrl)
+	}
+
+	query := localUrl.Query()
+
+	if query.Get(apiVersionQueryParameterName) != azureArcAPIVersion {
+		t.Fatalf("api-version not on %s got %s", azureArcAPIVersion, query.Get(apiVersionQueryParameterName))
+	}
+	if query.Get(resourceQueryParameterName) != strings.TrimSuffix(resource, "/.default") {
+		t.Fatal("suffix /.default was not removed.")
+	}
+	if result.Metadata.TokenSource != base.IdentityProvider {
+		t.Fatalf("expected IndenityProvider tokensource, got %d", result.Metadata.TokenSource)
+	}
+	if result.AccessToken != token {
+		t.Fatalf("wanted %q, got %q", token, result.AccessToken)
+	}
+	result, err = client.AcquireToken(context.Background(), resource)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if result.Metadata.TokenSource != base.Cache {
+		t.Fatalf("wanted cache token source, got %d", result.Metadata.TokenSource)
+	}
+	secondFakeClient, err := New(SystemAssigned(), WithHTTPClient(&mockClient))
+	if err != nil {
+		t.Fatal(err)
+	}
+	result, err = secondFakeClient.AcquireToken(context.Background(), resource)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if result.Metadata.TokenSource != base.Cache {
+		t.Fatalf("cache result wanted cache token source, got %d", result.Metadata.TokenSource)
+	}
+
+}
+
+func TestAzureArcOnlySystemAssignedSupported(t *testing.T) {
+	unsetEnvVars(t)
+	setEnvVars(t, AzureArc)
+	mockClient := mock.Client{}
+
+	setCustomAzureArcFilePath(t, fakeAzureArcFilePath)
+	for _, testCase := range []ID{
+		UserAssignedClientID("client"),
+		UserAssignedObjectID("ObjectId"),
+		UserAssignedResourceID("resourceid")} {
+		_, err := New(testCase, WithHTTPClient(&mockClient))
+		if err != nil {
```

Comment:
just a thought for simplifying this; remember `t.Fatal()` is tantamount to `return`
```go
if err == nil {
	t.Fatal(`expected error: azure arc not supported error"`)
}
if err.Error() != "azure Arc doesn't support user assigned managed identities" {
	t.Fatalf(`expected error: azure arc not supported error, got error: "%v"`, err)
}
```